soc/realtek/ameba/amebadplus/ameba_rom_symbol_acut_s.ld

```diff
@@ -0,0 +1,800 @@
+    PROVIDE(USBD_Reset = 0xebf1);
+    PROVIDE(USBD_SetSpeed = 0xec3d);
+    PROVIDE(USBD_Suspend = 0xec45);
+    PROVIDE(USBD_Resume = 0xec59);
+    PROVIDE(USBD_SOF = 0xec69);
+    PROVIDE(USBD_DevConnected = 0xec85);
+    PROVIDE(USBD_DevDisconnected = 0xec89);
+    PROVIDE(USBD_CtlSendData = 0xeca1);
+    PROVIDE(USBD_CtlPrepareRx = 0xecbd);
+    PROVIDE(USBD_CtlError = 0xecd9);
+    PROVIDE(USBD_SetupStage = 0xecf5);
+    PROVIDE(USBD_GetString = 0xf1d1);
+    PROVIDE(USBD_DebugInit = 0xf225);
+    PROVIDE(USBD_Debug = 0xf271);
+    PROVIDE(USBD_Debug_Config = 0xf2b5);
+    PROVIDE(USB_HalInit = 0xf2ed);
+    PROVIDE(USB_SetTurnaroundTime = 0xf3bd);
+    PROVIDE(USB_EnableGlobalInt = 0xf3e5);
+    PROVIDE(USB_DisableGlobalInt = 0xf3f9);
+    PROVIDE(USB_SetDeviceMode = 0xf40d);
+    PROVIDE(USB_EPPredictNextEp = 0xf435);
+    PROVIDE(USB_ResetLearnQueue = 0xf439);
+    PROVIDE(USB_FlushTxFifo = 0xf461);
+    PROVIDE(USB_FlushRxFifo = 0xf49d);
+    PROVIDE(USB_SetDevSpeed = 0xf4d5);
+    PROVIDE(USB_DevInit = 0xf4e5);
+    PROVIDE(USB_GetDevSpeed = 0xf625);
+    PROVIDE(USB_GetTxFifoNum = 0xf641);
+    PROVIDE(USB_ActivateEndpoint = 0xf651);
+    PROVIDE(USB_DeactivateEndpoint = 0xf6dd);
+    PROVIDE(USB_EP0StartXfer = 0xf745);
+    PROVIDE(USB_WritePacket = 0xf845);
+    PROVIDE(USB_EPStartXfer = 0xf86d);
+    PROVIDE(USB_ReadPacket = 0xfa01);
+    PROVIDE(USB_EPSetStall = 0xfa21);
+    PROVIDE(USB_EPClearStall = 0xfa5d);
+    PROVIDE(USB_StopDevice = 0xfa99);
+    PROVIDE(USB_SetDevAddress = 0xfae5);
+    PROVIDE(USB_DevConnect = 0xfb05);
+    PROVIDE(USB_DevDisconnect = 0xfb25);
+    PROVIDE(USB_ReadInterrupts = 0xfb45);
+    PROVIDE(USB_ReadDevAllOutEpInterrupt = 0xfb55);
+    PROVIDE(USB_ReadDevAllInEpInterrupt = 0xfb65);
+    PROVIDE(USB_ReadDevOutEPInterrupt = 0xfb75);
+    PROVIDE(USB_ReadDevInEPInterrupt = 0xfb8d);
+    PROVIDE(USB_ClearInterrupts = 0xfba5);
+    PROVIDE(USB_GetMode = 0xfbb5);
+    PROVIDE(USB_ActivateSetup = 0xfbc5);
+    PROVIDE(USB_EP0_OutStart = 0xfbfd);
+    PROVIDE(USBD_IsAttached = 0xff31);
+    PROVIDE(USBD_Process = 0xff3d);
+    PROVIDE(USBD_Init = 0xffb1);
+    PROVIDE(USBD_DeInit = 0xffed);
+    PROVIDE(USB_PCD_Init = 0xfff5);
+    PROVIDE(USB_PCD_Start = 0x101ed);
+    PROVIDE(USB_PCD_Stop = 0x10215);
+    PROVIDE(USB_PCD_DeInit = 0x10251);
+    PROVIDE(USB_PCD_DevConnect = 0x10291);
+    PROVIDE(USB_PCD_DevDisconnect = 0x102b5);
+    PROVIDE(USB_PCD_SetAddress = 0x102d9);
+    PROVIDE(USB_PCD_EP_Open = 0x10301);
+    PROVIDE(USB_PCD_EP_Close = 0x10379);
+    PROVIDE(USB_PCD_EP_Receive = 0x103cd);
+    PROVIDE(USB_PCD_EP_GetRxCount = 0x1041d);
+    PROVIDE(USB_PCD_EP_Transmit = 0x1042d);
+    PROVIDE(USB_PCD_EP_SetStall = 0x10481);
+    PROVIDE(USB_PCD_EP_ClrStall = 0x104e1);
+    PROVIDE(USB_PCD_EP_Flush = 0x1053d);
+    PROVIDE(USB_PCD_GetState = 0x10579);
+    PROVIDE(USB_PCD_SetTxFiFo = 0x10581);
+    PROVIDE(USB_PCD_SetRxFiFo = 0x105d9);
+    PROVIDE(USB_PCD_IRQHandler = 0x105e5);
```

Comment:
What is it and why is it there?

Reply:
amebadplus has a rom that contains some driver code, so it is the address of function in rom. shall I move it to hal_realtek repository?

Reply:
Why is it there and how do you think it could be used in the Zephyr RTOS? Did you understand my previous comment about the USB driver?

Reply:
1. amebadplus support trustzone, so cpu will run in secure world and non-secure world, here are two files in soc/realtek/ameba/amebadplus folder, ameba_rom_symbol_acut_s.ld and ameba_rom_symbol_acut.ld, ameba_rom_symbol_acut_s.ld is used in bootloader(like mucboot) and a loader for download image when mass production, there are all run in secure world; and ameba_rom_symbol_acut.ld is used in application image(such as freertos or zephyr), but for now we run zephyr in secure world, so you can see some code like usb/xmoderm, which is used for a loader in mass production. when we porting Trusted Firmware-M done, zephyr will run in non-secure world, and **the ameba_rom_symbol_acut.ld will be used, so the usb function in ameba_rom_symbol_acut_s.ld could not be used in the zephyr RTOS.**
2. your previous comment about the USB driver, here is reply from our team:
Dear Sir
There are two reasons why we can't use the DWC2 driver.
1) Our chips use a shared FIFO for all non-periodic (CTRL/BULK/INTR) IN endpoints, therefore, based on the DWC2 driver in the current zephyr SDK(3.7.0), our chips can not work because the DWC2 "only dedicated TX FIFO mode is supported".
We use the Shared NPTxFIFO, so we need to deal with the EPMIS interrupt and intoken learning queue, which are very important for us to complete the USB transfer.
2) the endpoint direction is limited in our chips, not all endpoints can support two-way transmission, for example, Endpoint 1 is only used for USB In transfers, Endpoint 2 is only used for USB Out transfers.

Thanks !

Reply:
>     1. Our chips use a shared FIFO for all non-periodic (CTRL/BULK/INTR) IN endpoints, therefore, based on the DWC2 driver in the current zephyr SDK(3.7.0), our chips can not work because the DWC2 "only dedicated TX FIFO mode is supported".
>        We use the Shared NPTxFIFO, so we need to deal with the EPMIS interrupt and intoken learning queue, which are very important for us to complete the USB transfer.
> 

You probably do not understand how to work on a common infrastructure. You are welcome to add support for the shared FIFO mode. Seriously, there will not be another driver for the same controller in Zephyr tree.

>     2. the endpoint direction is limited in our chips, not all endpoints can support two-way transmission, for example, Endpoint 1 is only used for USB In transfers, Endpoint 2 is only used for USB Out transfers.

I do not see any problems with that.

Reply:
I will discuss this with our team. but the symbol PROVIDE(USB_xxxx = 0xxxxx); is just in the ROM of the amebadplus asic which is used by the loader for mass production, and not used by zephyr. and this file was copied from our [github:](https://github.com/Ameba-AIoT/ameba-rtos/blob/master/amebadplus_gcc_project/project_km4/asdk/ld/ameba_rom_symbol_acut_s.ld) with no modification.
 Anyway, I have removed it for this PR.

Reply:
our team agree with what you said.

Reply:
Our team agrees with what you said and add the modification when commit usb dirver. thanks..